### PR TITLE
Adding scripts to run and monitor TPC-H/-DS queries at larger scale factors

### LINKED ISSUE #3455: Enhancement: Query suite script to run Accelerator TPCDS/H at Larger Scale Factors
_REMEMBER, BE **SMART**!_

_S: Specific_
_M: Measurable_
_A: Achievable_
_R: Relevant_
_T: Time-Bound_

## Goal-State/What/Result

We need a `workflow_dispatch` GitHub workflow that can run the TPC DS/H query suites at custom scale factors for Accelerator RC criteria

## Why/Purpose

As part of the RC criteria we have a requirement to run the query suites at larger scale factors to cover bases and to eliminate any possible issues when it comes memory usage or other possible issues at scale.

## By When

**Issue/Spec written and reviewed:** 11/12
**Done-Done:** 

## Done-Done

- [ ] [Principles Driven](https://github.com/spiceai/spiceai/blob/trunk/docs/PRINCIPLES.md)
- [ ] The Algorithm
- [ ] PM/Design Review
- [ ] DX/UX Review
- [ ] Threat Model / Security Review
- [ ] Tests
- [ ] Performance / Benchmarks
- [ ] Documentation
- [ ] Samples/Quickstarts/Tutorials
- [ ] Release Notes

## The Algorithm

- [ ] Every requirement questioned?
- [ ] Delete (Scope) any part you can.
- [ ] Simplify.
- [ ] Break down into smaller iterations/milestones.
- [ ] Opportunities for automation.

## Specification

_Provide a basic specification of the enhancement._

### Security Review

_Provide threat model and security review._

## How/Plan

_A clear and concise plan of how this should be implemented._

### Implementation (Sub-Issues)

- [ ] _Sub-Issue 1_
- [ ] _Sub-Issue 2_

### QA

- [ ] _Sub-Issue 1_
- [ ] _Sub-Issue 2_

### PR DESCRIPTION
**Pull request by eadgbear**
## 🗣 Description

Modifying and adding to @Sevenannn 's scripts to run the TPC suite queries. These scripts monitor CPU and memory usage and inject query names during the execution so they can be tracked alongside the metrics

## 🔨 Related Issues

- Closes #3455

### CODE REVIEW

File: test/scripts/setup-and-query.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+rm *.db 2> /dev/null
+
+# Cleanup spicepod.yaml
+rm spicepod.yaml 2> /dev/null
+
+subset_args=("${@:2}")
+
+# Setup the TPC-DS or TPC-H dataset, passing all args except the query folder
+./setup-tpc-spicepod.bash "${subset_args[@]}" &
+#./setup-tpc-spicepod.bash "${subset_args[@]}"
```
Comment (Sevenannn) [inline test/scripts/setup-and-query.sh]: Does this line need to be removed?

**Review by Sevenannn — COMMENTED** [review]
Nice changes! The monitor suite is really useful in terms of monitoring memory usage. I have some feedbacks on the structuring of the scripts. The current script looks like the combination of 2 of the following functionality. 
* setup
* run queries
* monitor & process results

However it’s a bit vague to me which script should be used. If user want to run queries, they will have run_queries_and_monitor, setup_and_query, run_queires scripts to choose from. Another example would be if user want to setup, run queries, and monitor and process results, it’s not very straightforward which combination of scripts they should use.
I'd suggest having the scripts for single functionality, and beyond that having 1 script that combines setup & run queries & monitor & process results together for a e2e testing purpose. lmk what you think regarding the structuing of the scripts

File: test/scripts/setup-and-query.sh (hunk 1 above)
Comment (eadgbear) [reply test/scripts/setup-and-query.sh]: Yes, you're absolutely right

**Review by Sevenannn — APPROVED** [review]
lgtm

**Pull request merged**